Code/Framework/AzToolsFramework/AzToolsFramework/ViewportSelection/ViewportEditorModeTracker.cpp

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzToolsFramework/API/ViewportEditorModeTrackerNotificationBus.h>
+#include <AzToolsFramework/ViewportSelection/ViewportEditorModeTracker.h>
+
+namespace AzToolsFramework
+{
+    static constexpr const char* ViewportEditorModeLogWindow = "ViewportEditorMode";
+
+    void ViewportEditorModes::SetModeActive(ViewportEditorMode mode)
+    {
+        if (const AZ::u32 modeIndex = static_cast<AZ::u32>(mode);
+            modeIndex < NumEditorModes)
+        {
+            m_editorModes[modeIndex] = true;
```

Comment:
comment: Aah I see we're using this array to know which modes are active or not, a mask is another option and I'm cautious about what happens if/when new modes are added, but in practice it's likely there weren't be many, it just means extending this is going to require more internal changes (slightly in conflict with the Open/Closed principle but for simplicity it might not actually be a problem in practice).

Reply:
👍